restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java

```diff
@@ -208,6 +144,133 @@ private Object convertToBackendBooleanValue(boolean value, boolean numericBoolea
+        unboundPaths[i++] = convertedPath;
+        leaf = convertedPath;
+      }
+    }
+
+    bindMap = addAllToBindMap(bindMap, unboundPaths, jsonValue, leaf, db.treatBooleansAsNumeric());
+    return new ImmutablePair<>(bindMap, firstLevelKeys);
+  }
+
+  private Map<String, Object> addAllToBindMap(
+      Map<String, Object> bindMap,
+      String[] unboundPaths,
+      JsonNode jsonValue,
+      String leaf,
+      boolean treatBooleansAsNumeric) {
+    bindMap = addUnboundPaths(bindMap, unboundPaths);
+    bindMap.put(DocumentDB.LEAF_VALUE_NAME, leaf);
+
+    if (jsonValue.isValueNode() && !jsonValue.isNull()) {
+      ValueNode value = (ValueNode) jsonValue;
+
+      if (value.isNumber()) {
+        bindMap.put(DocumentDB.DOUBLE_VALUE_NAME, value.asDouble());
+      } else if (value.isBoolean()) {
+        bindMap.put(
+            DocumentDB.BOOL_VALUE_NAME,
+            convertToBackendBooleanValue(value.asBoolean(), treatBooleansAsNumeric));
+      } else {
+        bindMap.put(DocumentDB.TEXT_VALUE_NAME, value.asText());
+      }
+    } else if (isEmptyObject(jsonValue)) {
+      bindMap.put(DocumentDB.TEXT_VALUE_NAME, DocumentDB.EMPTY_OBJECT_MARKER);
+    } else if (isEmptyArray(jsonValue)) {
+      bindMap.put(DocumentDB.TEXT_VALUE_NAME, DocumentDB.EMPTY_ARRAY_MARKER);
+    }
+    return bindMap;
+  }
+
+  private Map<String, Object> addUnboundPaths(Map<String, Object> bindMap, String[] unboundPaths) {
+    for (int i = 0; i < unboundPaths.length; i++) {
+      String unboundPath = unboundPaths[i];
+      if (unboundPath != null) {
+        bindMap.put("p" + i, unboundPath);
+      }
+    }
+    return bindMap;
+  }
+
+  private boolean isPatchingWithArrayValue(
+      int index, int pathSize, PathOperator op, boolean patching) {
+    return isAtTopLevel(index, pathSize) && op.getType() == PathOperator.Type.ARRAY && patching;
+  }
+
+  private boolean isAtTopLevel(int index, int pathSize) {
+    return index == pathSize;
+  }
+
+  private String convertPathValueForArrays(String innerPath, PathOperator op) {
+    String pathValue;
+    if (op.getType() == PathOperator.Type.ARRAY) {
+      int idx = Integer.parseInt(innerPath);
+      if (idx > docsApiConfiguration.getMaxArrayLength() - 1) {
+        throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_GENERAL_ARRAY_LENGTH_EXCEEDED);
+      }
+      // left-pad the array element to 6 characters
+      pathValue = "[" + DocsApiUtils.leftPadTo6(innerPath) + "]";
+    } else {
+      pathValue = innerPath;
+    }
+
+    return pathValue;
+  }
+
+  /**
+   * Takes a payload that is a URL-encoded form and turns it into data (bound parameter lists) to
+   * write to a collection. Also returns a list of all "top-level" keys in the form, for the
+   * purposes of merging key sets in the event of a PATCH operation. This is only used by certain
+   * performance tests, is not recommended for use, and will be deleted in the near future.
+   *
+   * @param db
```

Comment:
so this only reason this depends on the DocumentDB is:

* newBindMap
* treatBooleansAsNumeric

Can we extract bind map to some constants helper or whatever.. And have `treatBooleansAsNumeric` as parameter?

This way we can remove DocumentDB dependency completely..